src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -634,8 +628,15 @@ protected static void cleanUpSymbolicUnassembledEvents( final List<Haplotype> ha
         haplotypes.removeAll(haplotypesToRemove);
     }
 
-    protected static Map<Allele, List<Haplotype>> createAlleleMapper(final Map<VariantContext, Allele> mergeMap, final Map<Event, List<Haplotype>> eventMap ) {
-        return mergeMap.entrySet().stream().collect(Collectors.toMap(e -> e.getValue(), e -> eventMap.get(new Event(e.getKey())), (x,y)->x, LinkedHashMap<Allele, List<Haplotype>>::new));
+
+    protected static Map<Allele, List<Haplotype>> createAlleleMapper(List<VariantContext> eventsAtThisLoc, Map<Event, List<Haplotype>> eventMapper, VariantContext mergedVC) {
```

Comment:
My rule of thumb is use the `static` keyword whenever allowed, because why not.  I checked a few stackexchange threads, such as [this one](http://stackoverflow.com/questions/538870/should-private-helper-methods-be-static-if-they-can-be-static) and on the whole people agree.  As you said, `static` *does* signify that the code can be called from outside the class, but that's not necessarily a bad thing.  For example, it would make things much easier if anyone ever wanted to share the code with a new class or to extract it to a utility class.  For now, though, the `protected` modifier tells us that it's not currently used outside of this class.